TestProject/Controllers/PostController.cs

```diff
@@ -24,18 +24,40 @@ public PostController(IPostService postService)
         [HttpPost]
         [Route("createPost")]
         [CustomAuthorizationFilter]
-        public IActionResult Create([FromBody]PostEditorModel post)
+        public async Task<IActionResult> Create([FromBody]CreatePostModel post)
         {
-            _postService.Create(post, User.GetEmail());
+            await _postService.Create(post, User.GetEmail());
 
             return new StatusCodeResult(201);
         }
 
-        [Route("getAllPosts")]
-        public async Task<List<PostDisplayModel>> GetAllPostsAsync()
+        [Route("getPosts/{id?}")]
+        public async Task<List<PostDisplayModel>> GetPosts([FromRoute] int? id)
         {
-            var posts = await _postService.GetAllPostsAsync();
+            var posts = await _postService.GetPosts(id);
             return posts.Select(PostMapper.MapPostModelToPostModel).ToList();
         }
+
+        [Route("/posts/getPost/{id}")]
+        public async Task<EditPostModel> GetPost([FromRoute] int id)
+        {
+            return await _postService.Get(id);
+        }
+
+        [HttpPost]
+        [Route("/posts/editPost")]
+        public async Task<IActionResult> EditPost([FromBody] EditPostModel post)
```

Comment:
Remove `Post` suffix. You don't have it after `Delete` and `Create`. Be consistent at least. When we using `Post` controller, it's obvious, that we are dealing with posts. No need to be repetitive.